packages/features/insights/hooks/useDefaultRoutingForm.ts

```diff
@@ -0,0 +1,60 @@
+import { usePathname, useRouter } from "next/navigation";
+import { useEffect, useRef } from "react";
+
+import { useCompatSearchParams } from "@calcom/lib/hooks/useCompatSearchParams";
+import { trpc } from "@calcom/trpc";
+
+export function useDefaultRoutingForm({
+  teamId,
+  isAll,
+  routingFormId,
+  onRoutingFormChange,
+}: {
+  teamId: number | null | undefined;
+  isAll: boolean;
+  routingFormId: string | null | undefined;
+  onRoutingFormChange: (formId: string) => void;
+}) {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useCompatSearchParams();
+  const isRoutingInsights = pathname?.includes("/insights/routing");
+  const hasSetDefault = useRef(false);
+
+  // Query to get routing forms list
+  const { data: routingForms } = trpc.viewer.insights.getRoutingFormsForFilters.useQuery(
+    {
+      teamId: teamId ?? undefined,
+      isAll,
+    },
+    {
+      enabled: isRoutingInsights,
+    }
+  );
+
+  // Get the most popular form
+  const mostPopularForm = routingForms?.length
+    ? [...routingForms].sort((a, b) => (b._count?.responses ?? 0) - (a._count?.responses ?? 0))[0]
+    : null;
```

Comment:
NIT: Could we do this server side in the `getRoutingFormsForFilters` query? Maybe sort the forms by number of responses?

Reply:
We could be we already have the data here so it feels a bit redundant doing it in another endpoint when this just pulls from query cache